build_tools/cmake/iree_benchmark_suite.cmake

```diff
@@ -283,10 +283,35 @@ function(iree_benchmark_suite)
         DEPENDS "${_TOOL_FILE}"
       )
 
+      # Create the command and target for the command-line options spec used to
+      # compile the generated artifacts.
+      set(_COMPOPT_FILE "${_RUN_SPEC_DIR}/compilation_flagfile")
+      add_custom_command(
+        OUTPUT "${_COMPOPT_FILE}"
+        COMMAND
```

Comment:
I didn't find an easier way to dump multiple lines into a file, so using an extra Python script. Any other ideas?

Reply:
https://cmake.org/cmake/help/latest/command/file.html#writing or https://cmake.org/cmake/help/latest/command/configure_file.html#command:configure_file ?

Reply:
Thanks! Yes I've run into these two before.  Here we don't have key/value pairs like normal compiler defines and there are unknown number of different kinds of option strings to be emitted. So using `configure_file` would mean shorehorning for this specific case. For `file`, I didn't find a good way to wire it up with custom commands and targets. So not sure they are better than the current way..

Reply:
Hmm I think configure_file you can just sub in a list. The documentation mostly talks about cmakedefine, but it will replace `@VAR@` or `${VAR}` also. It seems like you should also be able to do something with cmake command mode:

```
add_custom_command(OUTPUT | TARGET ...
    COMMAND ${CMAKE_COMMAND} -E echo "<content-of-the-file>" > "</path/to/file>"
    )
```

but maybe the shell redirect would make it dependent on bash and mess things up on windows? Ugh it really seems like a python script shouldn't be necessary here, but maybe it's the least bad option

Reply:
I haven't looked at the details here, but I don't think we should need a 1 line python file (with 40 lines of supporting flag parsing) to generate a file :P

That would also place yet another fringe dep on python in the build system...

Reply:
Yeah that's my feeling as well :-D Also, is it important that this happen at build time instead of configure time? It seems like using `file(WRITE` and have it happen at configure time should be fine since it's all based on configure-time information

Reply:
The difficulty with `cmake -E echo` is that it cannot handle newlines. Giving it a string containing `\n` will cause it generate broken `build.ninja` file. :) 

On a second thought, I actually don't strictly need to use newline as separators. The compilation tools we have does not support `--flagfile` like the runtime tools we have anyway. So concatenating with  " " is fine for now at least. Helps to kick the can down the road a bit. Changed.

Reply:
Interesting. Didn't know that you can do `@file` to inject command-line options for LLVM binaries.. Okay that makes sense then. I switched to use `configure_file`. Thanks for the suggestion!